vercel/resource_project.go

```diff
@@ -206,11 +210,48 @@ At this time you cannot use a Vercel Project resource with in-line ` + "`environ
 							stringLengthBetween(1, 72),
 						},
 					},
-					"protect_production": schema.BoolAttribute{
-						Description: "If true, production deployments will also be protected",
-						Optional:    true,
-						Computed:    true,
-						Default:     booldefault.StaticBool(false),
+					"deployment_type": schema.StringAttribute{
+						Required:      true,
+						Description:   "The deployment environment to protect. Must be one of `standard_protection`, `all_deployments`, or `only_preview_deployments`.",
+						PlanModifiers: []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+						Validators: []validator.String{
+							stringOneOf("standard_protection", "all_deployments", "only_preview_deployments"),
+						},
+					},
+				},
+			},
+			"trusted_ips": schema.SingleNestedAttribute{
+				Description: "Ensures only visitors from an allowed IP address can access your deployment.",
+				Optional:    true,
+				Attributes: map[string]schema.Attribute{
+					"addresses": schema.SetAttribute{
+						Description:   "The allowed IP addressses and CIDR ranges with optional descriptions.",
+						Required:      true,
+						PlanModifiers: []planmodifier.Set{setplanmodifier.UseStateForUnknown()},
+						ElementType: types.ObjectType{
+							AttrTypes: map[string]attr.Type{
+								"value": types.StringType,
+								"note":  types.StringType,
+							},
+						},
+					},
+					"deployment_type": schema.StringAttribute{
+						Required:      true,
+						Description:   "The deployment environment to protect. Must be one of `standard_protection`, `all_deployments`, `only_production_deployments`, or `only_preview_deployments`.",
+						PlanModifiers: []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+						Validators: []validator.String{
+							stringOneOf("standard_protection", "all_deployments", "only_production_deployments", "only_preview_deployments"),
+						},
+					},
+					"protection_mode": schema.StringAttribute{
+						Optional:      true,
+						Computed:      true,
+						Default:       stringdefault.StaticString("additional"),
+						Description:   "Whether or not Trusted IPs is optional to access a deployment. Must be either `additional` or `exclusive`. `exclusive` is only available with Standalone Trusted IPs.",
+						PlanModifiers: []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+						Validators: []validator.String{
+							stringOneOf("additional", "exclusive"),
```

Comment:
We... probably need to document what `additional` and `exclusive` means here, since it differs in terminology to the UI quite a lot.

Reply:
Good idea, updated it 👍